### DISCUSSION TITLE
[Idea] Don't include default combination ID in product URL

### DISCUSSION THREAD

**Discussion by Amazzing** [PrestaShop/PrestaShop#32352] [Ideas]
### The problem

As you know, standard product URL includes **ID of product**, **ID of combination** and in most cases **#anchor** to selected attributes. Something like this: `/15-120-some-product.html#/1-size-s/10-color-red`

That is really useful for URLs pointing to **non-default** combinations. When a product page is accessed via URL with a combination identifier, required attributes are automatically pre-selected. 

But this rule also applies to URLs that are pointing to default combinations. However this is not required at all. 

Let's say combination **125** is the default combination of product **15**. 
In this case both of these links will point to exactly same page with same pre-selected attributes: 
`/15-125-some-product.html#/1-size-m/15-color-orange`
`/15-some-product.html`

Obviously 2nd link is better.
 
If you open any category page on a standard PS installation, you will notice, that most of product URLs include default combination identifiers that are useless. Once again - they are **useless for default combinations**, but they ar**e useful for non-default combinations**. 

### The solution

I am planning to submit a PR for removing default combination IDs from URLs. 
There can be different ways to achieve that behavior:
1. Modify `getProductURL()` in `ProductLazyArray.php`
```
if (!empty($product['cache_default_attribute'])
    && $product['cache_default_attribute'] == $product['id_product_attribute']) {
    $canonical = true;
}

return $this->link->getProductLink(
    $product['id_product'],
    ...
```
This fix will work for most of themes. But it will update URLs everywhere, including cart summary.
Probably default combination identifiers should remain in cart summary. 

2. Override `getURL()` only for `ProductListingLazyArray.php`
Same as above, it will update URLs in cart summary

3. Modify `/miniatures/product.tpl` in **classic** theme 
```
{if !empty($product.cache_default_attribute)
    && $product.cache_default_attribute == $product.id_product_attribute}
    {$product_url = $product.canonical_url}
{else}
    {$product_url = $product.url}
{/if}
{* and then use $product_url instead of $product.url *}
```
This fix will work only on actual product listing pages like category, new-products, etc...
But many existing shops will not benefit from this update, because most of them use custom themes, and not all of them update their themes regularly.  Moreover, it can take a while before custom theme developers update their tpls. 

So, what do you think would be the most optimal way to go? 

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by camlafit** [comment]
Hello

Yes could be nice to restore old behavior about default product url, I don't see any use where id_product_attribute could be useful.
I've searched on history to understand we this change was done but without success.

I presume was to simplify router/dispatcher rules, but without conviction.

Have a nice day
.
